docs/development/monitoring-stack.md

```diff
@@ -0,0 +1,192 @@
+
+Basic instruction how to define alert rules can be found in the [Prometheus documentation](https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules).
+
+### Routing tree
+The Alertmanager is grouping incoming alerts based on labels into buckets. Each bucket have its own configuration like alert receivers, initial delaying duration or resending frequency etc. You can find more informaton about Alertmanager routing in the [Prometheus/Alertmanager documentation](https://prometheus.io/docs/alerting/configuration/#route). The routing trees for the Alertmanagers deployed by Gardener are depicted bellow.
+
+Central Seed Alertmanager
+```
+∟ main route (all alerts for all shoots on the seed will enter)
+  ∟ group by project and shoot name
+    ∟ group by severity "blocker" and "critical" → route to Garden operators
+    ∟ group by severity "warning" (dropped)
+```
+
+Shoot Alertmanager
+```
+∟ main route (only alerts for one Shoot will enter)
+  ∟ group by severity "blocker" and "critical" → route to cluster alert receiver
+  ∟ group by severity "warning" (dropped, will change soon → route to cluster alert receiver)
+```
+
+All alerts related to components running on the Shoot workers are inhibited in case of an issue with the vpn connection, because those components can't be scraped anymore and Prometheus will fire alerts in consequence. The components running on the workers are probably healthy and the alerts are presumably false positives.
+
+### Alert Attributes
+Each alert rule definition have to contain the following annotations:
+- **summary**: A short description of the issue.
+- **description**: A detailed explanation of the issue with hints to the possible root causes and the impact assessment of the issue.
+
+In addtion each alert must contain the following labels:
+- **type**
+  - `shoot`: Components running on the Shoot worker nodes in the `kube-system` namespace.
+  - `seed`: Components running on the Seed in the Shoot namespace as part of/next to the control plane.
+- **service**
+  - Name of the component (in lowercase) e.g. `kube-apiserver`, `alertmanager` or `vpn`.
+- **severity**
+  - `blocker`: All issues which make the cluster entirely unusable e.g. `KubeAPIServerDown` or `KubeSchedulerDown`
+  - `critical`: All issues which affect single functionalities/components but not affect the cluster in its core functionality e.g. `VPNDown` or `KubeletDown`.
+  - `warning`: No current existing issue, rather a hint for situations which could lead to real issue in the close future e.g. `HighLatencyApiServerToWorkers` or `ApiServerResponseSlow`.
+
+### Alert tests
+All alerts should come along with tests. Not yet defined how to integrate those tests.
```

Comment:
Let's remove this section about tests if we cannot provide information here right now?

Reply:
Yep done. We will add the section later when we have defined how to integrate the alerts tests.